### PR TITLE
[Merged by Bors] - feat: Port CategoryTheory.Adjunction.Opposites

### PR DESCRIPTION
**Pull request by adamtopaz**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
Lint done.

File: Mathlib/CategoryTheory/Adjunction/Opposites.lean [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+#align
+  category_theory.adjunction.adjoint_of_op_adjoint_op
+  CategoryTheory.Adjunction.adjointOfOpAdjointOp
+
+-- Porting note: LHS of these simplify to below
+attribute [-simp, nolint simpNF] adjointOfOpAdjointOp_homEquiv_apply 
+  adjointOfOpAdjointOp_homEquiv_symm_apply
+
+-- Porting note: simpNF reports simplifying to itself here
+@[simp, nolint simpNF]
```
Comment (ChrisHughes24) [inline Mathlib/CategoryTheory/Adjunction/Opposites.lean:62]: I think the linter was right to flag this theorem. The LHS is simplified to something that looks the same, but has some different implicit arguments somewhere. Consequently, a lemma with the exact same statement as this one is not proved by `simp` even with this lemma marked as a `simp` lemma. I think this theorem is pretty useless until the LHS is in normal form including implicits.

File: Mathlib/CategoryTheory/Adjunction/Opposites.lean [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+namespace CategoryTheory.Adjunction
+
+/-- If `G.op` is adjoint to `F.op` then `F` is adjoint to `G`. -/
+@[simps!]
+def adjointOfOpAdjointOp (F : C ⥤ D) (G : D ⥤ C) (h : G.op ⊣ F.op) : F ⊣ G :=
+  Adjunction.mkOfHomEquiv {
+    homEquiv := fun {X Y} =>
+      ((h.homEquiv (Opposite.op Y) (Opposite.op X)).trans (opEquiv _ _)).symm.trans
+        (opEquiv _ _)
+    homEquiv_naturality_left_symm := by
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Adjunction/Opposites.lean:43]: I would have hope there was a porting note here noting the regression that this proof is needed at all!

**Comment by semorrison** [comment]
The problematic lemmas generated by `simps` (which `simpNF` is correctly flagging) are not actually used anywhere (verified by backporting to mathlib3), so the simplest solution is to only generated the `simp` lemmas we need in the first place. I've backported their removal in https://github.com/leanprover-community/mathlib/pull/18680.

File: Mathlib/CategoryTheory/Adjunction/Opposites.lean (hunk 2 above)
Comment (adamtopaz) [reply Mathlib/CategoryTheory/Adjunction/Opposites.lean:47]: Done!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**